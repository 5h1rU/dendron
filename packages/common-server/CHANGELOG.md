# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.24](https://github.com/dendronhq/dendron/compare/v0.3.23...v0.3.24) (2020-07-20)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.21](https://github.com/dendronhq/dendron/compare/v0.3.20...v0.3.21) (2020-07-19)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.20](https://github.com/dendronhq/dendron/compare/v0.3.19...v0.3.20) (2020-07-18)

### Bug Fixes

- handle special characters in file names when importing using LocalFilePod ([03e42b1](https://github.com/dendronhq/dendron/commit/03e42b167ac9f073cd56f10c4e31b5cecf66dabf))

## [0.3.19](https://github.com/dendronhq/dendron/compare/v0.3.18...v0.3.19) (2020-07-18)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.18](https://github.com/dendronhq/dendron/compare/v0.3.17...v0.3.18) (2020-07-18)

### Features

- ability to import notes from local file system [docs](https://github.com/dendronhq/dendron-template/blob/master/vault/dendron.topic.pod.md) ([e44fd11](https://github.com/dendronhq/dendron/commit/e44fd1133bec10f22831f059c8d98cf4076dcdcc))

## [0.3.17](https://github.com/dendronhq/dendron/compare/v0.3.16...v0.3.17) (2020-07-17)

### Features

- more flexible file names for [journal notes](https://github.com/dendronhq/dendron-template/blob/master/vault/dendron.special-notes.md) ([9d9f10b](https://github.com/dendronhq/dendron/commit/9d9f10bd0873c201361c4625fc49c9f62ee82991))

## [0.3.16](https://github.com/dendronhq/dendron/compare/v0.3.15...v0.3.16) (2020-07-16)

### Features

- implement journal notes. see details here: https://github.com/dendronhq/dendron-template/blob/master/vault/dendron.feature.journals.md ([5e1236f](https://github.com/dendronhq/dendron/commit/5e1236fddbf1e0fddf4c27d1a40e9841cc99974f))

## [0.3.15](https://github.com/dendronhq/dendron/compare/v0.3.14...v0.3.15) (2020-07-14)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.11](https://github.com/dendronhq/dendron/compare/v0.3.10...v0.3.11) (2020-07-13)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.10](https://github.com/dendronhq/dendron/compare/v0.1.6...v0.3.10) (2020-07-13)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.9](https://github.com/dendronhq/dendron/compare/v0.3.8...v0.3.9) (2020-07-09)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.8](https://github.com/dendronhq/dendron/compare/v0.3.7...v0.3.8) (2020-07-09)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.7](https://github.com/dendronhq/dendron/compare/v0.3.6...v0.3.7) (2020-07-08)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.6](https://github.com/dendronhq/dendron/compare/v0.3.5...v0.3.6) (2020-07-07)

**Note:** Version bump only for package @dendronhq/common-server

# [0.3.0](https://github.com/dendronhq/dendron/compare/v0.2.20...v0.3.0) (2020-07-05)

**Note:** Version bump only for package @dendronhq/common-server
