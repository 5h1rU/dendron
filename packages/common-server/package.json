{
  "name": "@dendronhq/common-server",
  "version": "0.1.6",
  "description": "common-server",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "ssh://git@github.com/dendronhq/dendron.git",
    "directory": "packages/common-server"
  },
  "author": {
    "name": "Kevin Lin",
    "email": "kevin@thence.io",
    "url": "https://kevinslin.com",
    "organization": true
  },
  "keywords": [
    ""
  ],
  "files": [
    "lib"
  ],
  "main": "lib/index",
  "types": "lib/index",
  "scripts": {
    "clean": "rimraf lib && rimraf coverage",
    "format": "",
    "lint": "",
    "prepublishOnly": "npm run build",
    "prebuild": "npm run clean && npm run format && npm run lint && echo Using TypeScript && tsc --version",
    "build": "npm run compile",
    "compile": "tsc -p tsconfig.build.json",
    "test": "jest",
    "coverage": "jest --coverage",
    "watch": "npm run compile -- --watch",
    "test:watch": "jest --watch"
  },
  "dependencies": {
    "@dendronhq/common-all": "^0.1.6",
    "gray-matter": "^4.0.2",
    "lodash": "^4.17.15",
    "minimatch": "^3.0.4",
    "pino": "^6.3.2",
    "yamljs": "^0.3.0"
  },
  "devDependencies": {
    "@types/pino": "^6.3.0",
    "@types/jest": "^23.3.3",
    "@types/lodash": "^4.14.154",
    "@types/minimatch": "^3.0.3",
    "@types/node": "^10.11.4",
    "@types/yamljs": "^0.2.31",
    "coveralls": "^3.0.2",
    "jest": "^23.6.0",
    "prettier": "^1.14.3",
    "rimraf": "^2.6.2",
    "ts-jest": "^23.10.3",
    "ts-node": "^7.0.1",
    "tslint": "^5.11.0",
    "tslint-config-prettier": "^1.15.0",
    "typescript": "^3.1.1"
  },
  "engines": {
    "node": ">=10.0.0"
  },
  "jest": {
    "preset": "ts-jest"
  },
  "gitHead": "403eb012f8b34bc115dd7dd86f80c76b39329f3b"
}
