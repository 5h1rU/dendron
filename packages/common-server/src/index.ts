export * from "./files";
export * from "./logger";
export { testUtils, FileTestUtils } from "./testUtils";
