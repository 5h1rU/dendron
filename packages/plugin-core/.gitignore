out
node_modules
.vscode-test/
*.vsix
build
notes
reports
assets/LAST_COMMIT
assets/dendronWS
OVSX_PAT
dist
fixtures/fake-workspace