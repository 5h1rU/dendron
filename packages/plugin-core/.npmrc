registry=https://npm.pkg.github.com/dendronhq
