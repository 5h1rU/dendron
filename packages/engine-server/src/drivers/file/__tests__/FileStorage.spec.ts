import {
  AssertionError,
  IDNode,
  QueryMode,
  Schema
} from "@dendronhq/common-all";
import {
  createFileStorage,
  expectSnapshot,
  readMdFile,
  setupTmpDendronDir
} from "../../../testUtils";

import FileStorage from "../store";
import _ from "lodash";
import fs from "fs-extra";
import { FileTestUtils } from "packages/common-server";

describe("main", () => {
  let root: string;
  let store: FileStorage;

  beforeEach(() => {
    root = setupTmpDendronDir();
    store = createFileStorage(root);
  });

  afterEach(() => {
    fs.removeSync(root);
  });

  describe("notes", () => {
    let queryMode: QueryMode;

    beforeAll(() => {
      queryMode = "note";
    });

    describe("delete", () => {
      test("delete foo.one", async () => {
        const resp = await store.query("**/*", queryMode, {});
        const nodeOrig = _.find(
          resp.data,
          n => n.title === "foo.one"
        ) as IDNode;
        await store.delete(nodeOrig.id);
        await expect(store.get(nodeOrig.id, {})).rejects.toThrow(
          AssertionError
        );
        expect(() => {
          readMdFile(root, "foo.one.md");
        }).toThrow(/ENOENT/);
      });
    });

    describe("get", () => {
      test("get root", async () => {
        const resp = await store.query("**/*", queryMode, {});
        const nodeOrig = _.find(resp.data, n => n.title === "root") as IDNode;
        const respGet = await store.get(nodeOrig.id, {});
        const node = respGet.data;
        expect(node).not.toBeUndefined();
        expect(node.fname).toEqual("root");
      });

      test("get foo.one", async () => {
        const resp = await store.query("**/*", queryMode, {});
        const nodeOrig = _.find(
          resp.data,
          n => n.title === "foo.one"
        ) as IDNode;
        const respGet = await store.get(nodeOrig.id, {});
        const node = respGet.data;
        expect(node).not.toBeUndefined();
        expect(node.fname).toEqual("foo.one");
      });
    });

    /**
     * Hiearchy:
     * - foo
     *   - foo.one
     *   - foo.two
     */
    describe("query", () => {
      test("with missing", async () => {
        FileTestUtils.writeMDFile(
          root,
          "bar.one.alpha.md",
          {},
          "bar alpha content"
        );
        store = createFileStorage(root);
        const resp = await store.query("**/*", queryMode, {});
        const bar = _.find(resp.data, { fname: "bar" });
        const barOne = _.find(resp.data, { fname: "bar.one" });
        expect(bar).not.toBeNull();
        expect(bar?.stub).toBeTruthy();
        expect(barOne).not.toBeNull();
        expect(barOne?.stub).toBeTruthy();
        expectSnapshot(expect, "missing snapshots", resp.data);
      });

      test("all", async () => {
        const resp = await store.query("**/*", queryMode, {});
        const rootNode = _.find(resp.data, n => n.title === "root") as Schema;
        const foo = _.find(resp.data, n => n.title === "foo") as Schema;
        const fooChild = _.find(
          resp.data,
          n => n.title === "foo.one"
        ) as Schema;
        expect(rootNode).not.toBeUndefined();
        expect(fooChild).not.toBeUndefined();
        // expect parent
        expect(fooChild.domain.title).toEqual("foo");
        expect(fooChild.parent?.id).toEqual(foo.id);
        expectSnapshot(expect, "raw-props", fooChild);
      });
    });

    describe("write", () => {
      test("writeQuery", async () => {
        const resp = await store.query("**/*", queryMode, {});
        const node = _.find(resp.data, n => n.title === "foo.one") as IDNode;
        node.body = "bond";
        await store.write(node);
        const { data, content } = readMdFile(root, "foo.one.md");
        expect(data.title).toEqual("foo.one");
        expect(content).toEqual("bond\n");
        expect(content).toMatchSnapshot("data snapshot");
      });
    });
  });

  describe("schema", () => {
    let queryMode: QueryMode;

    beforeAll(() => {
      queryMode = "schema";
    });

    describe("query", () => {
      test("all", async () => {
        const resp = await store.query("**/*", queryMode, {});
        const rootNode = _.find(resp.data, n => n.title === "root") as IDNode;
        const foo = _.find(resp.data, n => n.title === "foo") as IDNode;
        const node = _.find(resp.data, n => n.title === "one") as IDNode;
        expect(rootNode).not.toBeUndefined();
        expect(node).not.toBeUndefined();
        expect(node.parent?.id).toEqual(foo.id);
      });
    });

    describe("write", () => {
      test("writeRoot", async () => {
        await store.write(Schema.createRoot());
        const data = FileTestUtils.readYMLFile(root, "root.schema.yml");
        expect(data).toMatchSnapshot("data snapshot");
      });
    });
  });
});
