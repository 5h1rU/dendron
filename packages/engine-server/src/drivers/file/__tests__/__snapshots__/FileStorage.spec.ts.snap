// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main notes query all: raw-props 1`] = `
Object {
  "body": "",
  "children": Array [],
  "custom": Object {
    "bond": 42,
  },
  "data": Object {},
  "desc": "",
  "fname": "foo.one",
  "parent": "foo",
  "stub": false,
  "title": "foo.one",
  "type": "note",
}
`;

exports[`main notes query with missing: missing snapshots 1`] = `
Array [
  Object {
    "body": "",
    "children": Array [
      "foo",
      "bar",
    ],
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "root",
    "parent": "root",
    "stub": false,
    "title": "root",
    "type": "note",
  },
  Object {
    "body": "",
    "children": Array [
      "foo.one",
      "foo.two",
      "foo.three",
    ],
    "custom": Object {},
    "data": Object {
      "schemaId": "b111db5b-bc52-4977-893b-307522f89ea3",
    },
    "desc": "",
    "fname": "foo",
    "parent": "root",
    "stub": false,
    "title": "foo",
    "type": "note",
  },
  Object {
    "body": "",
    "children": Array [
      "bar.one",
    ],
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "bar",
    "parent": "root",
    "stub": true,
    "title": "bar",
    "type": "note",
  },
  Object {
    "body": "",
    "children": Array [],
    "custom": Object {
      "bond": 42,
    },
    "data": Object {},
    "desc": "",
    "fname": "foo.one",
    "parent": "foo",
    "stub": false,
    "title": "foo.one",
    "type": "note",
  },
  Object {
    "body": "",
    "children": Array [],
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "foo.two",
    "parent": "foo",
    "stub": false,
    "title": "foo.two",
    "type": "note",
  },
  Object {
    "body": "",
    "children": Array [
      "foo.three.alpha",
    ],
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "foo.three",
    "parent": "foo",
    "stub": true,
    "title": "foo.three",
    "type": "note",
  },
  Object {
    "body": "",
    "children": Array [
      "bar.one.alpha",
    ],
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "bar.one",
    "parent": "bar",
    "stub": true,
    "title": "bar.one",
    "type": "note",
  },
  Object {
    "body": "",
    "children": Array [],
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "foo.three.alpha",
    "parent": "foo.three",
    "stub": false,
    "title": "foo.three.alpha",
    "type": "note",
  },
  Object {
    "body": "bar alpha content
",
    "children": Array [],
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "bar.one.alpha",
    "parent": "bar.one",
    "stub": false,
    "title": "bar.one.alpha",
    "type": "note",
  },
]
`;

exports[`main notes write writeQuery: data snapshot 1`] = `
"bond
"
`;

exports[`main schema write writeRoot: root 1`] = `
Object {
  "body": "",
  "custom": Object {},
  "data": Object {},
  "desc": "",
  "fname": "root.schema",
  "stub": false,
  "title": "root",
  "type": "schema",
}
`;
